### PR TITLE
a little fix

### PR DESCRIPTION
**Pull request by KLIM-nk01**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -29,9 +29,7 @@ app.use("/profile", profileRoutes);
 
 start = async () => {
     try {
-     
         await mongoose.connect(config.get("mongoToken"));
-
```
Comment (katsiaryna-savchuk-itechart) [inline app.js:34]: why removing empty lines?

File: controllers/profile.js [hunk 2]
```diff
@@ -1,26 +1,51 @@
...
+
+            const userData = {
+                name: patient.userData.name,
+                lastName: patient.userData.lastName,
+                photo: patient.userData.photo,
+                phone: patient.userData.phone,
+                email: patient.userData.email,
+                visit: visits,
+                birthday: patient.birthday,
+                address: patient.address,
```
Comment (katsiaryna-savchuk-itechart) [inline controllers/profile.js:45]: the keys of these two objects are exactly the same, what's the point of listing them?

File: controllers/profile.js [hunk 3]
```diff
@@ -37,23 +62,194 @@ exports.profileDoctorGet = async (req, res) => {
...
+                phone: doctor.userData.phone,
+                email: doctor.userData.email,
+                experience: doctor.experience,
+                direction: doctor.direction,
+                workPlace: {
+                    name: doctor.workPlace.name,
+                    address: doctor.workPlace.address,
+                },
+                workTime: workTime,
+                patients: doctor.patients,
```
Comment (katsiaryna-savchuk-itechart) [inline controllers/profile.js:96]: same as above here

File: controllers/profile.js [hunk 4]
```diff
@@ -37,23 +62,194 @@ exports.profileDoctorGet = async (req, res) => {
...
+            };
+            res.status(200).json(doctorData);
+        }
+    } catch (e) {
+        console.log(e);
+        res.send({ message: "Server error" });
```
Comment (katsiaryna-savchuk-itechart) [inline controllers/profile.js:102]: message is repeated with the same string, makes sense to make it a reusable constant

File: controllers/profile.js [hunk 5]
```diff
@@ -37,23 +62,194 @@ exports.profileDoctorGet = async (req, res) => {
...
+};
+
+exports.profileDoctorAddDatePut = async (req, res) => {
+    try {
+        const doctor = await Doctor.findOne({
+            userData: req.user.id,
+        });
+        if (doctor) {
+            const newDateTime = await new DateTime({
+                doctorData: doctor.id,
```
Comment (katsiaryna-savchuk-itechart) [inline controllers/profile.js:113]: why doctor data equals to id, it's a bit confusing

File: controllers/profile.js [hunk 6]
```diff
@@ -37,23 +62,194 @@ exports.profileDoctorGet = async (req, res) => {
...
+                date: req.body.date.split(" ")[0],
+                time: req.body.date.split(" ")[1],
+            });
+            await Doctor.findOneAndUpdate(
+                {
+                    userData: req.user.id,
```
Comment (katsiaryna-savchuk-itechart) [inline controllers/profile.js:119]: same as above

File: controllers/profile.js [hunk 7]
```diff
@@ -37,23 +62,194 @@ exports.profileDoctorGet = async (req, res) => {
...
+                {
+                    $push: { workTime: newDateTime.id },
+                }
+            );
+            await newDateTime.save();
+            res.json(newDateTime);
+        }
+    } catch (e) {
+        console.log(e);
+        res.send({ message: "Server error" });
```
Comment (katsiaryna-savchuk-itechart) [inline controllers/profile.js:130]: to const

File: controllers/profile.js [hunk 8]
```diff
@@ -37,23 +62,194 @@ exports.profileDoctorGet = async (req, res) => {
...
+    }
+};
+
+exports.profileDoctorDeleteDataDelete = async (req, res) => {
+    try {
+        const { idDate } = req.params;
+        const user = await Doctor.findOne({
+            userData: req.user.id,
```
Comment (katsiaryna-savchuk-itechart) [inline controllers/profile.js:138]: same as above

File: controllers/profile.js [hunk 9]
```diff
@@ -37,23 +62,194 @@ exports.profileDoctorGet = async (req, res) => {
...
+            workTime: idDate,
+        });
+        if (user) {
```
Comment (katsiaryna-savchuk-itechart) [inline controllers/profile.js:141]: pls add empty lines for better readability

File: controllers/profile.js [hunk 10]
```diff
@@ -37,23 +62,194 @@ exports.profileDoctorGet = async (req, res) => {
...
+                            patientData: user.id,
+                        },
+                    }
+                );
+                const response = {
+                    doctorName: date.doctorData.userData.name,
+                    doctorLastName: date.doctorData.userData.lastName,
+                    doctorDirection: date.doctorData.direction,
+                    medCenterName: date.doctorData.workPlace.name,
+                    medCenterAddress: date.doctorData.workPlace.address,
```
Comment (katsiaryna-savchuk-itechart) [inline controllers/profile.js:212]: use destructure

File: controllers/profile.js [hunk 11]
```diff
@@ -37,23 +62,194 @@ exports.profileDoctorGet = async (req, res) => {
...
+                    time: date.time,
+                    date: date.date,
+                };
+
+                res.json(response);
+            }
+        }
+    } catch (e) {
+        console.log(e);
+        res.send({ message: "Server error" });
```
Comment (katsiaryna-savchuk-itechart) [inline controllers/profile.js:222]: to const

File: controllers/profile.js [hunk 12]
```diff
@@ -37,23 +62,194 @@ exports.profileDoctorGet = async (req, res) => {
...
+    }
+};
+exports.profilePatientDeleteAppointmentDelete = async (req, res) => {
+    try {
+        const { idDate } = req.params;
+        const user = await Patient.findOne({
+            userData: req.user.id,
```
Comment (katsiaryna-savchuk-itechart) [inline controllers/profile.js:229]: confusing names

**Review by katsiaryna-savchuk-itechart — CHANGES_REQUESTED** [review]
pls check the comments

File: controllers/auth.js [hunk 13]
```diff
@@ -16,20 +17,50 @@ exports.authPost = async (req, res) => {
...
+            })
+        );
     } catch (e) {
         console.log(e);
         res.send({ message: "Server error" });
     }
 };
+
+exports.refreshTokens = (refreshToken) => {
+    let payload = jwt.verify(refreshToken, config.get("secretKey"));
```
Comment (katsiaryna-savchuk-itechart) [inline controllers/auth.js:39]: why using `let` instead of `const`?

File: controllers/profile.js [hunk 14]
```diff
@@ -1,61 +1,251 @@
...
+                _idDate: visitItem._id,
+            }));
+
+            const userData = {
+                name: patient.userData.name,
+                lastName: patient.userData.lastName,
+                photo: patient.userData.photo,
+                phone: patient.userData.phone,
+                email: patient.userData.email,
+                // ...patient.userData,
```
Comment (katsiaryna-savchuk-itechart) [inline controllers/profile.js:44]: remove comment

File: controllers/profile.js [hunk 15]
```diff
@@ -1,61 +1,251 @@
...
+
+                res.json(response);
+            }
+        }
+    } catch (e) {
+        console.log(e);
+        res.send({ message: constants.ERRORS_MESSAGE.SERVER_ERROR });
+    }
+};
+exports.profilePatientDeleteAppointmentDelete = async (req, res) => {
```
Comment (katsiaryna-savchuk-itechart) [inline controllers/profile.js:228]: add lines between the controllers for better readability

File: controllers/registration.js [hunk 16]
```diff
@@ -71,23 +71,22 @@ exports.registrationUserPost = async (req, res) => {
...
-            user: {
-                id: user.id,
-                name: user.name,
-                userRole: user.userRole,
-                photo: user.photo,
-            },
-        });
+        updateTokens(user.id).then((tokens) =>
+            res.status(200).send({
+                tokens,
```
Comment (katsiaryna-savchuk-itechart) [inline controllers/registration.js:81]: why sending tokens in response body?

File: middleware/auth.middleware.js [hunk 17]
```diff
@@ -1,20 +1,44 @@
...
+                });
+                
+                req.user = {
+                    userId: userIdFromRefreshToken.userId,
+                };
+                next();
+                return;
+            }
+        } else if (e instanceof jwt.JsonWebTokenError) {
+            res.status(401).json({ message: "Invalid token!" });
```
Comment (katsiaryna-savchuk-itechart) [inline middleware/auth.middleware.js:40]: what for sending token error to the client?

File: routes/refreshToken.routes.js [hunk 18]
```diff
@@ -0,0 +1,7 @@
+const express = require("express");
+const router = express.Router();
+const authController = require("../controllers/auth");
+
+router.get("/", authController.refreshTokens);
```
Comment (katsiaryna-savchuk-itechart) [inline routes/refreshToken.routes.js:5]: refresh token should be handled through middleware, not as a separate route

**Review by katsiaryna-savchuk-itechart — CHANGES_REQUESTED** [review]
check the comment + clean the code

File: app.js (hunk 1 above)
Comment (KLIM-nk01) [reply app.js:34]: add in all files

File: controllers/profile.js (hunk 10 above)
Comment (KLIM-nk01) [reply controllers/profile.js:215]: server send trash to the server

File: controllers/profile.js (hunk 14 above)
Comment (KLIM-nk01) [reply controllers/profile.js:44]: removed

File: routes/refreshToken.routes.js (hunk 18 above)
Comment (KLIM-nk01) [reply routes/refreshToken.routes.js:5]: removed that rout

File: controllers/auth.js [hunk 19]
```diff
@@ -16,20 +17,50 @@ exports.authPost = async (req, res) => {
...
+    try {
+        if (payload.type !== "refresh") {
+            res.status(400).json({ message: "Invalid Token!" });
+            return;
+        }
+    } catch (e) {
+        if (e instanceof jwt.TokenExpiredError) {
+            res.status(400).json({ message: "Token expired!" });
+            return;
+        } else if (e instanceof jwt.JsonWebTokenError) {
```
Comment (katsiaryna-savchuk-itechart) [inline controllers/auth.js:50]: what's the point of else if statement here?

File: controllers/auth.js [hunk 20]
```diff
@@ -16,20 +17,50 @@ exports.authPost = async (req, res) => {
...
+            res.status(400).json({ message: "Invalid Token!" });
```
Comment (katsiaryna-savchuk-itechart) [inline controllers/auth.js:51]: no need to send message about tokens to the client

File: controllers/profile.js (hunk 15 above)
Comment (KLIM-nk01) [reply controllers/profile.js:240]: fixed

File: controllers/profile.js (hunk 12 above)
Comment (KLIM-nk01) [reply controllers/profile.js]: discussed and decided to leave

File: controllers/auth.js (hunk 19 above)
Comment (KLIM-nk01) [reply controllers/auth.js:65]: removed all else and add return to if

File: controllers/auth.js (hunk 20 above)
Comment (KLIM-nk01) [reply controllers/auth.js:66]: changed to err 500 and message: "Server Error" or "Unauthorized"

File: controllers/registration.js (hunk 16 above)
Comment (KLIM-nk01) [reply controllers/registration.js]: fixed, access and refresh token send to the client in cookies now

File: controllers/auth.js [hunk 21]
```diff
@@ -1,35 +1,87 @@
...
+            res.status(401).json({ message: UNAUTHORIZED });
+            return;
+        }
+    } catch (e) {
+        if (e instanceof jwt.TokenExpiredError) {
+            res.status(401).json({ message: UNAUTHORIZED });
+            return;
+        }
+        if (e instanceof jwt.JsonWebTokenError) {
+            res.status(401).json({ message: UNAUTHORIZED });
```
Comment (katsiaryna-savchuk-itechart) [inline controllers/auth.js:71]: in all cases you return the same response from the server, why not just log the error and send a message to the client? I mean without `if` statements

File: controllers/profile.js [hunk 22]
```diff
@@ -1,61 +1,275 @@
...
+                populate: {
+                    path: "doctorData",
+                    select: ["direction", "workPlace", "userData"],
+
+                    populate: [
+                        { path: "workPlace", select: ["name", "address"] },
+                        { path: "userData", select: ["name", "lastName"] },
+                    ],
+                },
+            });
```
Comment (katsiaryna-savchuk-itechart) [inline controllers/profile.js:25]: add empty line pls

**Review by katsiaryna-savchuk-itechart — APPROVED** [review]
Pls fix minor format comments and add `public` folder to gitignore before merging the pr

**Pull request merged**